src/middlewared/middlewared/alert/service/telegram.py

```diff
@@ -0,0 +1,26 @@
+import requests
+
+from middlewared.alert.base import ThreadedAlertService
+from middlewared.schema import Dict, Str
+
+
+class TelegramAlertService(ThreadedAlertService):
+    title = "Telegram"
+
+    schema = Dict(
+        "Telegram_attributes",
+        Str("base_url", default="https://api.telegram.org/bot"),
+        Str("TELEGRAM_CHAT_ID", required=True, empty=False),
+        Str("TELEGRAM_BOT_TOKEN", required=True, empty=False),
+        strict=True,
+    )
+
+    def send_sync(self, alerts, gone_alerts, new_alerts):
+        base_url = self.attributes["base_url"] or "https://api.Telegram.com/bot"
+	url = {base_url} + {self.attributes['TELEGRAM_BOT_TOKEN']} + '/sendMessage'
+	post_data = {"chat_id": {self.attributes['TELEGRAM_CHAT_ID']}, "text": self._format_alerts(alerts, gone_alerts, new_alerts), parse_mode=Markdown}
```

Comment:
Markdown is not a valid name.

Reply:
please check the sendmessage api docs [here](https://core.telegram.org/bots/api#sendmessage)
> parse_mode | String | Optional | Send Markdown or HTML, if you want Telegram apps to show bold, italic, fixed-width text or inline URLs in your bot's message.

and the sample is [here](https://core.telegram.org/bots/api#markdown-style)

Reply:
You have not tested this code, as you said yourself. Markdown is a variable, not defined in this scope.

Reply:
Also there is a mix between spaces and tabs.

Reply:
i see ur point .. should be fixed